library/Versions.re

```diff
@@ -145,3 +146,20 @@ let getRemoteVersions = () => {
      )
   |> Lwt.return;
 };
+
+let throwIfInstalled = versionName => {
+  switch(getInstalledVersions()) {
+  | Ok(x) => {
+    Array.to_list(x)
+    |> List.exists(x => {
+      open Local;
+      x.name == versionName;
+    })
+    |> x => switch x {
+    | true => Lwt.fail(Already_installed(versionName));
+    | false => Lwt.return();
+    };
+  };
+  | Error(_) => Lwt.return();
```

Comment:
I think it'll be more readable if the short paths will be on the top of the switch cases, so they would feel like early returns:

```reason
switch (x) {
  | Error(_) => Lwt.return()
  | Ok(o) => {
       ...
  }
}
```